addons/sourcemod/scripting/royale/config.sp

```diff
@@ -117,16 +117,37 @@ void Confg_GetMapFilepath(char[] filePath, int length)
 	GetCurrentMap(mapName, sizeof(mapName));
 	GetMapDisplayName(mapName, mapName, sizeof(mapName));
 	
+	int partsCount = CountCharInString(mapName, '_') + 1;
+	//The map has no prefix/suffix?
+	if (partsCount < 2)
+	{
+		BuildPath(Path_SM, filePath, length, "configs/royale/maps/%s.cfg", mapName);
+		return;
+	}
+	
 	//Split map prefix and first part of its name (e.g. pl_hightower)
-	char nameParts[2][PLATFORM_MAX_PATH];
-	ExplodeString(mapName, "_", nameParts, sizeof(nameParts), sizeof(nameParts[]));
+	char[][] nameParts = new char[partsCount][PLATFORM_MAX_PATH];
+	ExplodeString(mapName, "_", nameParts, partsCount, PLATFORM_MAX_PATH);
 	
-	//Stitch name parts together
+	//Start to stitch name parts together
 	char tidyMapName[PLATFORM_MAX_PATH];
-	Format(tidyMapName, sizeof(tidyMapName), "%s_%s", nameParts[0], nameParts[1]);
-	
-	//Build file path
+	Format(tidyMapName, PLATFORM_MAX_PATH, "%s", nameParts[0]);
```

Comment:
Use `strcopy` if there is no actual formatting going on, and pass `sizeof(tidyMapName)`.